### ISSUE TITLE
Add log event streams for jobs, tasks and taskflows

### ISSUE THREAD

**Issue by cjh1** [Kitware/cumulus#230]
We current have a server side event stream for clusters:

```
GET /clusters/:id/log/stream
```

The idea is to add similar endpoints to jobs, tasks and taskflows.

This would involve factoring out this [block of code](https://github.com/Kitware/cumulus/blob/master/girder/cumulus/server/cluster.py#L468-L505) into a common module so it can be reused. Probably move into cumulus.common.girder

The one concern with this approach is we are going to proliferate the number of event streams we have open. We probably do some consolidation and have:

```
GET /clusters/log/stream
GET /jobs/log/stream
GET /tasks/log/stream
GET /taskflows/log/stream
```

@TristanWright I think we need to discuss our approach a little more, but you can start looking at factoring out the code block ....

**Comment by TristanWright** [comment]
To start I'm abstracting a block to `cumulus/cumulus/common/girder.py`

```py
def stream_gen(log_count, timeout, records):
    import cherrypy
    import time
    # defaults taken from Girder
    MIN_POLL_INTERVAL = 0.5
    MAX_POLL_INTERVAL = 2

    cherrypy.response.headers['Content-Type'] = 'text/event-stream'
    cherrypy.response.headers['Cache-Control'] = 'no-cache'

    def streamGen():
        start = time.time()
        wait = MIN_POLL_INTERVAL
        last_log_seen = log_count  # prevents UnboundLocalError

        while cherrypy.engine.state == cherrypy.engine.states.STARTED:
            wait = min(wait + MIN_POLL_INTERVAL, MAX_POLL_INTERVAL)

            for i, logMessage in enumerate(records):
                last_log_seen += 1
                wait = MIN_POLL_INTERVAL
                start = time.time()
                yield sseMessage(logMessage)

            if time.time() - start > timeout:
                break

        time.sleep(wait)

    return streamGen
```

Tests for the cluster log stream aren't working anymore, I believe I have all the right imports. Will ask tomorrow.

**Comment by cjh1** [comment]
@TristanWright Let talk when you get in, I am also looking at the core Girder notification stream to see if we could add something there that we could reuse.

**Comment by cjh1** [comment]
Fixed in #231
